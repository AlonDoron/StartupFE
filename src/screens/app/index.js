export { default as Home } from "./Home";
export { default as Account } from "./Account";
export { default as MyServices } from "./MyServices";
