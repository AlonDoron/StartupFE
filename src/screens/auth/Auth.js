import React from "react";
import { View } from "react-native";
import { Button } from "react-native-paper";
import i18n from "../../i18n";

const Auth = ({ navigation }) => {
  return (
    <View>
      <View>
        <Button
          style={{ margin: "5%" }}
          mode="contained"
          onPress={() => navigation.navigate("Login")}
        >
          {i18n.t("login")}
        </Button>
      </View>

      <View>
        <Button
          style={{ margin: "5%" }}
          mode="contained"
          onPress={() => navigation.navigate("Registration")}
        >
          {i18n.t("registration")}
        </Button>
      </View>
    </View>
  );
};

export default Auth;
