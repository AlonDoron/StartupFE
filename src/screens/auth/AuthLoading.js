import React, { useEffect } from "react";
import { StatusBar, View } from "react-native";
import { ActivityIndicator } from "react-native-paper";
import { ApiConfig } from "../../config";
import HttpClient from "../../api/HttpClient";
import TokensHandler from "../../api/TokensHandler";

const AuthLoading = (props) => {
  useEffect(() => {
    fetchToken().then((token) => {
      let params = {
        userId: token,
      };

      if (!params.userId) props.navigation.navigate("Auth");
      else
        HttpClient.get(ApiConfig.IDENTITY_PORT, "api/identity", params)
          .then((result) => {
            props.navigation.navigate(result ? "App" : "Auth");
          })
          .catch((err) => console.log(err));
    });
  }, []);

  const fetchToken = async () => {
    return TokensHandler.getTokenFromDevice();
  };

  return (
    <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
      <ActivityIndicator size="large" />
      <StatusBar barStyle="default" />
    </View>
  );
};

export default AuthLoading;
