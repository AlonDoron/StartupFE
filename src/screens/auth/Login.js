import React, { useEffect, useState } from "react";
import { View } from "react-native";
import LoginForm from "../../forms/LoginForm";
import { useDispatch, useSelector } from "react-redux";
import { isUserExistsByPhoneNumber } from "../../actions/authAction";

const Login = (props) => {
  const [values, setValues] = useState({});
<<<<<<< HEAD
  const [allowNavigate, setAllowNavigate] = useState(false);

  const dispatch = useDispatch();
  const isUserExists = useSelector((state) => state.auth.isUserExists);

  const handleSubmitForm = (vals) => {
    setValues(vals);
    dispatch(isUserExistsByPhoneNumber("Login", vals)).then(() =>
      setAllowNavigate(true)
    );
  };

  useEffect(() => {
    if (allowNavigate)
=======
  const dispatch = useDispatch();
  const isUserExists = useSelector((state) => state.auth.isUserExists);
  const isFetching = useSelector((state) => state.auth.isFetching);
  const isDoneFetching = useSelector((state) => state.auth.isDoneFetching);

  const handleSubmitForm = (vals) => {
    dispatch(isUserExistsByPhoneNumber("Login", vals));
    setValues(vals);
  };

  useEffect(() => {
    if (isDoneFetching)
>>>>>>> 7ed5aadfe5115e279991c9bfc2a26b243ed84fa7
      props.navigation.navigate(isUserExists ? "VerifyAuth" : "Signup", {
        vals: values,
        sentFrom: "Login",
      });
<<<<<<< HEAD
  }, [allowNavigate, isUserExists, values]);
=======
  }, [isDoneFetching]);
>>>>>>> 7ed5aadfe5115e279991c9bfc2a26b243ed84fa7

  return (
    <View>
      <LoginForm submitForm={handleSubmitForm} submitting={isFetching} />
    </View>
  );
};

export default Login;
