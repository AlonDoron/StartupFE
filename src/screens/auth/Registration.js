import React, { useEffect, useState } from "react";
import { StyleSheet } from "react-native";
import { Layout } from "@ui-kitten/components";
import { RegistrationForm } from "../../forms";
import { useDispatch, useSelector } from "react-redux";
import { isUserExistsByPhoneNumber } from "../../actions/authAction";

const Registration = ({ navigation }) => {
  const [values, setValues] = useState({});
  const [allowNavigate, setAllowNavigate] = useState(false);

  const dispatch = useDispatch();
  const isUserExists = useSelector((state) => state.auth.isUserExists);
  const isFetching = useSelector((state) => state.auth.isFetching);

  const handleSubmitForm = (vals) => {
    setValues(vals);
    dispatch(
      isUserExistsByPhoneNumber("Registration", {
        phoneNumber: vals.phoneNumber,
      })
    ).then(() => setAllowNavigate(true));
  };

  useEffect(() => {
    if (allowNavigate) {
      if (isUserExists) navigation.navigate("Login");
      else
        navigation.navigate("VerifyAuth", {
          vals: values,
          sentFrom: "Registration",
        });
    }
  }, [allowNavigate]);

  return (
    <Layout style={styles.container}>
      <RegistrationForm submitForm={handleSubmitForm} submitting={isFetching} />
    </Layout>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    alignContent: "center",
  },
});

export default Registration;
