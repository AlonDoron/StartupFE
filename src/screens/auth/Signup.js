import React, { useState } from "react";
import { View } from "react-native";
import SignupForm from "../../forms/SignupForm";
import HttpClient from "../../api/HttpClient";
import apiConfig from "../../config/apiConfig";

const Signup = (props) => {
  const [submitting, setSubmitting] = useState(false);

  const handleSubmitForm = (vals) => {
    setSubmitting(true);

    HttpClient.get(
      apiConfig.REGISTRATION_PORT,
      "api/Registration/checkIfUserExists",
      { phoneNumber: vals.phoneNumber }
    )
      .then((result) => {
        setSubmitting(false);
        props.navigation.navigate(result ? "VerifySignup" : "Auth", {
          ...vals,
        });
      })
      .catch((e) => console.log(e));
  };

  return (
    <View>
      <SignupForm submitForm={handleSubmitForm} submitting={submitting} />
    </View>
  );
};

export default Signup;
