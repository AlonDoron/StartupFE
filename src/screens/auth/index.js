export { default as AuthLoading } from "./AuthLoading";
export { default as Auth } from "./Auth";
export { default as Login } from "./Login";
export { default as Signup } from "./Signup";
export { default as VerifyLogin } from "./VerifyLogin";
export { default as VerifySignup } from "./VerifySignup";
