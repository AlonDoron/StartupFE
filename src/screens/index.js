import {default as LandPageScreen} from './LandPageScreen';
import {default as LoginViaPhoneScreen} from './LoginViaPhoneScreen';
import {default as PhoneNumberValidation} from './PhoneNumberValidation';
import {default as PhoneNumberValidationTest} from './PhoneNumberValidationTest';
import {default as RegistrationScreen} from './RegistrationScreen';
