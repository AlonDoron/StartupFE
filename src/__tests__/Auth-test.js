import React from "react";
import { Auth } from "../screens/auth";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<Auth />).toJSON();
  expect(tree).toMatchSnapshot();
});
