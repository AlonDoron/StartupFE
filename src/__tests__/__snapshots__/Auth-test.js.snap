// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "margin": "5%",
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "margin": "5%",
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Registration
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
